src/amo/components/LandingPage.js

```diff
@@ -41,9 +44,33 @@ export class LandingPageBase extends React.Component {
 
   static defaultProps = {
     apiAddonType: getApiAddonType,
+    contentForType: null,
   }
 
-  contentForType(visibleAddonType) {
+  componentWillMount() {
+    this.setViewContextType();
+  }
+
+  componentDidUpdate() {
+    this.setViewContextType();
+  }
+
+  setViewContextType() {
+    const { apiAddonType, dispatch, params } = this.props;
+
+    try {
+      const addonType = apiAddonType(params.visibleAddonType);
+      dispatch(setViewContext(addonType));
+    } catch (err) {
+      if (err instanceof AddonTypeNotFound) {
+        log.info('AddonTypeNotFound in setViewContextType()', err);
```

Comment:
Because of the way the router works the app can hit this page instead of a 404, thinking getting `addonType` as whatever is in the URL after `/en-US/firefox/` so `/en-US/firefox/foo/` passes addonType `foo`.

#2161 should really help with that...